Index: src/Libraries/Hyena.Gui/Hyena.Data.Gui/ListView/ListView_Rendering.cs
===================================================================
--- src/Libraries/Hyena.Gui/Hyena.Data.Gui/ListView/ListView_Rendering.cs	(revision 5192)
+++ src/Libraries/Hyena.Gui/Hyena.Data.Gui/ListView/ListView_Rendering.cs	(working copy)
@@ -67,7 +67,7 @@
             
             base.OnStyleSet (old_style);
             RecomputeRowHeight = true;
-            theme = new GtkTheme (this);
+            theme = Hyena.Gui.Theming.ThemeEngine.CreateTheme (this);
 
             // Save the drawable so we can reuse it
             Gdk.Drawable drawable = cell_context != null ? cell_context.Drawable : null;
Index: src/Libraries/Hyena.Gui/Hyena.Gui.Theming/ThemeEngine.cs
===================================================================
--- src/Libraries/Hyena.Gui/Hyena.Gui.Theming/ThemeEngine.cs	(revision 0)
+++ src/Libraries/Hyena.Gui/Hyena.Gui.Theming/ThemeEngine.cs	(revision 0)
@@ -0,0 +1,67 @@
+// 
+// ThemeEngine.cs
+//  
+// Author:
+//   Aaron Bockover <abockover@novell.com>
+// 
+// Copyright 2009 Aaron Bockover
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using System;
+
+namespace Hyena.Gui.Theming
+{
+    internal class SuperHackThemeEngineProviderArgs : EventArgs
+    {
+        private Gtk.Widget widget;
+        public Gtk.Widget Widget {
+            get { return widget; }
+            set { widget = value; }
+        }
+    
+        private Theme theme;
+        public Theme Theme {
+            set { theme = value; }
+            get { return theme; }
+        }
+    }
+
+    public static class ThemeEngine
+    {   
+        private static EventHandler provider;
+        
+        public static void SetProvider (EventHandler provider)
+        {
+            ThemeEngine.provider = provider;
+        }
+        
+        public static Theme CreateTheme (Gtk.Widget widget)
+        {
+            if (provider == null) {
+                return new GtkTheme (widget);
+            }
+            
+            SuperHackThemeEngineProviderArgs args = new SuperHackThemeEngineProviderArgs ();
+            args.Widget = widget;
+            provider (null, args);
+            return args.Theme ?? new GtkTheme (widget);
+        }
+    }
+}
Index: src/Libraries/Hyena.Gui/Hyena.Gui.csproj
===================================================================
--- src/Libraries/Hyena.Gui/Hyena.Gui.csproj	(revision 5192)
+++ src/Libraries/Hyena.Gui/Hyena.Gui.csproj	(working copy)
@@ -29,7 +29,7 @@
     <Reference Include="System" />
     <Reference Include="gtk-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
     <Reference Include="gdk-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
-    <Reference Include="Mono.Cairo, Version=1.0.5000.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756" />
+    <Reference Include="Mono.Cairo" />
     <Reference Include="pango-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
     <Reference Include="glib-sharp, Version=2.12.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f" />
     <Reference Include="System.Xml" />
@@ -127,13 +127,13 @@
     <Compile Include="Hyena.Gui.Theatrics\Pulsator.cs" />
     <Compile Include="Hyena.Gui\PixbufImageSurface.cs" />
     <Compile Include="Hyena.Widgets\TextViewEditable.cs" />
+    <Compile Include="Hyena.Gui.Theming\ThemeEngine.cs" />
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
   <ProjectExtensions>
     <MonoDevelop>
       <Properties>
         <Deployment.LinuxDeployData generateScript="false" />
-        <GtkDesignInfo />
         <MonoDevelop.Autotools.MakefileInfo IntegrationEnabled="true" RelativeMakefileName="Makefile.am">
           <BuildFilesVar Sync="true" Name="SOURCES" />
           <DeployFilesVar />
Index: src/Libraries/Hyena.Gui/Hyena.Widgets/RoundedFrame.cs
===================================================================
--- src/Libraries/Hyena.Gui/Hyena.Widgets/RoundedFrame.cs	(revision 5192)
+++ src/Libraries/Hyena.Gui/Hyena.Widgets/RoundedFrame.cs	(working copy)
@@ -92,7 +92,7 @@
         protected override void OnRealized ()
         {
             base.OnRealized ();
-            theme = new GtkTheme (this);
+            theme = Hyena.Gui.Theming.ThemeEngine.CreateTheme (this);
         }
 
         protected override void OnSizeRequested (ref Requisition requisition)
Index: src/Libraries/Hyena.Gui/Hyena.Widgets/MessageBar.cs
===================================================================
--- src/Libraries/Hyena.Gui/Hyena.Widgets/MessageBar.cs	(revision 5192)
+++ src/Libraries/Hyena.Gui/Hyena.Widgets/MessageBar.cs	(working copy)
@@ -111,7 +111,7 @@
         protected override void OnRealized ()
         {
             base.OnRealized ();
-            theme = new GtkTheme (this);
+            theme = Hyena.Gui.Theming.ThemeEngine.CreateTheme (this);
         }
         
         protected override void OnSizeAllocated (Gdk.Rectangle allocation)
Index: src/Libraries/Hyena.Gui/Makefile.am
===================================================================
--- src/Libraries/Hyena.Gui/Makefile.am	(revision 5192)
+++ src/Libraries/Hyena.Gui/Makefile.am	(working copy)
@@ -40,6 +40,7 @@
 	Hyena.Gui.Theming/GtkTheme.cs \
 	Hyena.Gui.Theming/Theme.cs \
 	Hyena.Gui.Theming/ThemeContext.cs \
+	Hyena.Gui.Theming/ThemeEngine.cs \
 	Hyena.Gui/CairoExtensions.cs \
 	Hyena.Gui/CleanRoomStartup.cs \
 	Hyena.Gui/CompositeUtils.cs \
